app/scripts/controllers/metamask-notifications/metamask-notifications.ts

```diff
@@ -490,16 +641,16 @@ export class MetamaskNotificationsController extends BaseController<
    *
    * **Action** - used in the notifications settings page to enable/disable snap notifications.
    *
-   * @param state - A boolean value indicating the desired enabled state of the snap notifications.
+   * @param snapNotificationsEnabled - A boolean value indicating the desired enabled state of the snap notifications.
    * @async
    * @throws {Error} If the BearerToken token or storage key is missing.
    */
-  public async setSnapNotificationsEnabled(state: boolean) {
+  public async setSnapNotificationsEnabled(snapNotificationsEnabled: boolean) {
     try {
       this.#assertAuthEnabled();
```

Comment:
Does this means the need to authenticate to have snaps notifications enabled ?

Reply:
I'm reworking this in a separate PR (this is too large lol).

Does the snaps controller/existing controllers have logic to enable/disable snaps?

The idea of this `metamask-notifications` controller was to be a single source of truth/state for all things notifications (you are right, this is tied quite heavily to the authentication/wallet notifications work).

Reply:
In fact I don't think we want to do any authentication for this method. I'll remove!

Reply:
As discussed, this and the snap switch have been removed :)